### ISSUE TITLE
isWifiOn() versus isOnline() for wifi menu 

### ISSUE THREAD

**Issue by v01d** [koreader/koreader#4192]
I'm currently adapting Koreader to work on a Sony PRS-T2, running Debian instead of the stock android (will make a PR soon with that). During the porting process, I had some issues with how the WiFi status is reported in the menu. It seems that the menu considers WiFi to be on if the device can reach the internet (check via isOnline()) instead of actually checking if the WiFi is up (via isWifiOn()). In my case, since I could be plugged in via USB and this makes a network-over-usb connection, it makes the device "online" however that is not the result of having WiFi connected.
Moreover, even if WiFi is up, it may not be connected (no IP received, unreachable internet) and koreader will consider WiFi off and offer putting WiFi up again. I think this relates to #2706.

For my port-in-progress, I have made this change, which I guess won't be acceptable if I make the PR:

<pre>
--- a/frontend/ui/network/manager.lua
+++ b/frontend/ui/network/manager.lua
@@ -121,10 +121,10 @@ end
 function NetworkMgr:getWifiMenuTable()
     return {
         text = _("Wi-Fi connection"),
-        enabled_func = function() return Device:isAndroid() or Device:isKindle() or Device:isKobo() end,
-        checked_func = function() return NetworkMgr:isOnline() end,
+        enabled_func = function() return Device:isAndroid() or Device:isKindle() or Device:isKobo() or Device:isPRST() end,
+        checked_func = function() return NetworkMgr:isWifiOn() end,
         callback = function(menu)
-            local wifi_status = NetworkMgr:isOnline()
+            local wifi_status = NetworkMgr:isWifiOn()
             local complete_callback = function()
                 -- notify touch menu to update item check state
                 menu:updateItems()
</pre>

With this change, it behaves as expected but I'm not sure if I'm breaking other things or if it actually makes sense (to me it does). My question is: should I condition the change on Device:isPRST()? Or is this actually something that should be set like this for all devices?

**Comment by Frenzie** [comment]
Btw, you can use:
`````
```diff
`````

Note that there might be some ugly side-effects related to reverting #2383 just like that, see #2355. But I agree that probably wasn't the best fix.

I'm not really seeing a way someone could end up with an empty info popup with the way things are today. It looks like #2394 by @poire-z probably implements a proper fix as a side effect, meaning your proposed reversal of #2383 above should be perfectly acceptable in its present form.
https://github.com/koreader/koreader/blob/75df3fcea3fa717dafe5993ea6604e705b8624c0/frontend/device/generic/device.lua#L199-L228

**Comment by v01d** [comment]
I think there are some changes in #2383 which are OK, there are some things that should depend on isOnline() (such as when attempting to reach the internet). I was only mentioning the wifi menu, which is supposed to control your wifi connection (ie: allow to put the interface down even if it is not connected to the internet).

Ok, so I will go ahead with the change as-is and make it part of the PR. Or do you prefer a separate PR for this?

**Comment by Frenzie** [comment]
I think it's generally be clearer to keep distinct things separate, but if you're the type of person who submits a well-structured set of commits that shouldn't be squashed, tacking it onto the same PR shouldn't hurt if it's easier for you. ;-)

**Comment by v01d** [comment]
I think you can close this now.

**Comment by v01d** [comment]
Yeah, but I didn't know if you wanted it open as a reminder for taking care of this more extensively.
